docker/mint/mint.Dockerfile

```diff
@@ -31,11 +70,41 @@ RUN pip3 install -r /libra/docker/mint/requirements.txt
 
 RUN mkdir -p /opt/libra/bin  /libra/client/data/wallet/
 
-COPY --from=builder /libra/target/release/cli /opt/libra/bin
-COPY --from=builder /libra/target/release/config-builder /opt/libra/bin
+ARG TARGET_DIR=/tmp/dockertarget
+COPY --from=builder ${TARGET_DIR}/release/cli /opt/libra/bin
+COPY --from=builder ${TARGET_DIR}/release/config-builder /opt/libra/bin
 COPY docker/mint/server.py /opt/libra/bin
 COPY docker/mint/docker-run.sh /opt/libra/bin
 
+###########################################################################
+#  Test the pre-production image before shipping
+###########################################################################
+FROM pre-test AS test
+
+# This test is racy as it relies on side-effects (in this case the pid file's existence.
+# If the minter shuts down before the pid file can be accessed, then the test will fail.
+# Currently the minter is trying to access validator nodes and take several seconds to
+# time out.
+
+# Necessary environment variabls for gunicorn (in docker-run.sh)
+# Setup and perform the test
+# GIVEN
+ARG IMAGE="libra_mint:latest"
+ARG AC_HOST="172.18.0.13"
+ARG AC_PORT="8000"
+ARG LOG_LEVEL="info"
+ARG MINT_KEY="1337133713371337133713371337133713371337"
+# WHEN
+RUN /opt/libra/bin/docker-run.sh & \
+   while [ ! -f /tmp/pidfile ]; do sleep 1; done; \
+   PIDS=$( cat /tmp/pidfile ); \
```

Comment:
I'm wonder if there is another way to find the PID of gunicorn without writing it to file in `docker-run.sh`.

Reply:
My typical ps -ef | grep tool wasn't on the path, and was reluctant to add it - but can for the test.
Not sure why we need gunicorn today.   Meaning to read through our python server file at some point soon.

Reply:
hmm... I think it's worth to add those basic utils. Just not a fan of writing PID to file.